pkg/controller/attach_detach.go

```diff
@@ -312,59 +314,59 @@ func (r *ReconcileAttachDetach) triggerDetach(ctx context.Context, azVolumeAttac
 	var err error
 	ctx, w := workflow.New(ctx)
 	defer func() { w.Finish(err) }()
-	// only detach if detachment request was made for underlying volume attachment object
-	detachmentRequested := volumeDetachRequested(azVolumeAttachment)
 
-	if detachmentRequested {
-		defer r.stateLock.Delete(azVolumeAttachment.Name)
-		if _, ok := r.stateLock.LoadOrStore(azVolumeAttachment.Name, nil); ok {
-			return getOperationRequeueError("detach", azVolumeAttachment)
-		}
+	defer r.stateLock.Delete(azVolumeAttachment.Name)
+	if _, ok := r.stateLock.LoadOrStore(azVolumeAttachment.Name, nil); ok {
+		return getOperationRequeueError("detach", azVolumeAttachment)
+	}
 
-		updateFunc := func(obj client.Object) error {
-			azv := obj.(*azdiskv1beta2.AzVolumeAttachment)
-			// Update state to detaching
-			_, derr := updateState(azv, azdiskv1beta2.Detaching, normalUpdate)
-			return derr
+	updateFunc := func(obj client.Object) error {
+		azv := obj.(*azdiskv1beta2.AzVolumeAttachment)
+		// Update state to detaching
+		updateMode := normalUpdate
+		if r.isDriverUninstall() {
```

Comment:
We use state transition as a way to prevent multiple goroutines working on same tasks. If we use the `force-update` mode here, we will end up with multiple goroutines requesting detach for the same node and volume pair.